### PR TITLE
Emergency report

### PR DESCRIPTION
**Pull request by samshara**
## Changes
- Add emergency report section

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/EmergencyReportAndDocument/index.tsx [hunk 1]
```diff
@@ -1,17 +1,316 @@
...
+    });
+
+    const {
+        pending: appealDocumentsPending,
+        response: appealDocumentsResponse,
+    } = useRequest({
+        skip: isNotDefined(emergencyResponse?.appeals)
+           || (isDefined(emergencyResponse)
+            && isDefined(emergencyResponse.appeals)
+            && emergencyResponse.appeals.length < 1),
```
Comment (tnagorra) [inline src/views/EmergencyReportAndDocument/index.tsx]: Let's not mix || and &&

File: src/views/EmergencyReportAndDocument/index.tsx [hunk 2]
```diff
@@ -1,17 +1,316 @@
...
+            appeal: emergencyResponse.appeals.map((appeal) => appeal.id).filter(isDefined),
+            limit: PAGE_SIZE,
+            offset: PAGE_SIZE * (page - 1),
+        } : undefined, // TODO: fix typing issue in server
+    });
+
+    const getRegionList = useCallback((regionList: number[]) => (
+        regionList.map((region) => regions?.find((r) => r.id === region))
+            .filter(isDefined)
+    ), [regions]);
```
Comment (tnagorra) [inline src/views/EmergencyReportAndDocument/index.tsx]: Let's create a map instead and directly get the object

File: src/views/EmergencyReportAndDocument/index.tsx [hunk 3]
```diff
@@ -1,17 +1,316 @@
...
+                (item) => item.countries,
+            ),
+            createRegionListColumn<FieldReportListItem, number>(
+                'regions',
+                strings.fieldReportsTableRegion,
+                (item) => {
+                    const ids = unique(item.countries.map((country) => country.region))
+                        .filter(isDefined);
+                    return getRegionList(ids);
+                },
```
Comment (tnagorra) [inline src/views/EmergencyReportAndDocument/index.tsx]: This is not very good for performance. Should we instead update the data on the table?

cc: @frozenhelium

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/EmergencyReportAndDocument/index.tsx (hunk 3 above)
Comment (frozenhelium) [reply src/views/EmergencyReportAndDocument/index.tsx]: Yes, and we can also use `createListDisplayColumn` which has similar use case

File: src/views/EmergencyReportAndDocument/index.tsx [hunk 4]
```diff
@@ -1,17 +1,331 @@
...
+import TextOutput from '#components/TextOutput';
+import useRegion, { type Region } from '#hooks/domain/useRegion';
 import useTranslation from '#hooks/useTranslation';
+import { adminUrl } from '#config';
+import { numericIdSelector } from '#utils/selectors';
+import { resolveToString } from '#utils/translation';
+import { resolveUrl } from '#utils/resolveUrl';
+import { useRequest } from '#utils/restRequest';
+import type { EmergencyOutletContext } from '#utils/outletContext';
+import type { GoApiResponse } from '#utils/restRequest';
```
Comment (tnagorra) [inline src/views/EmergencyReportAndDocument/index.tsx]:
```suggestion
import { type EmergencyOutletContext } from '#utils/outletContext';
import { type GoApiResponse } from '#utils/restRequest';
```

File: src/views/EmergencyReportAndDocument/index.tsx [hunk 5]
```diff
@@ -1,17 +1,331 @@
...
+           || ((isDefined(emergencyResponse)
+            && isDefined(emergencyResponse.appeals)
+            && emergencyResponse.appeals.length < 1)),
+        url: '/api/v2/appeal_document/',
+        query: isDefined(emergencyResponse) ? {
+            appeal: emergencyResponse.appeals.map((appeal) => appeal.id).filter(isDefined),
+            limit: PAGE_SIZE,
+            offset: PAGE_SIZE * (page - 1),
+        } : undefined, // TODO: fix typing issue in server
+    });
```
Comment (tnagorra) [inline src/views/EmergencyReportAndDocument/index.tsx]: Instead of sending appeal, we should send "emergency" id to the server.

Comment (tnagorra) [reply src/views/EmergencyReportAndDocument/index.tsx]: Let's add a FIXME

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**